### PR TITLE
Feature/UI

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
The first commit adds a simple sidebar for choosing the visualization of specific generations. The subsequent commits add a settings page, a settings class, window switching capability, threading, and a stop button for quitting the application prematurely. Other commits involve adding a custom file watcher, an image loader, and selection from listbox. The pull request also includes updates to the DynamicData and a descriptive README. The changes made aim to improve the usability and functionality of the GeneticDFA application.

### CODE REVIEW

File: GeneticDFA/Setup.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        Settings settings = settingsService.LoadSettings();
+
+        string testTracePath = $"{Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)}/traces.json";
+
+        List<TestTrace> traces = DFAUtility.ImportTestTraces(testTracePath);
+        List<char> alphabet = DFAUtility.DiscoverAlphabet(traces).ToList();
+
+        double fitnessUpperBound = settings.RewardTruePositive * traces.Count(t => t.IsAccepting) +
+                                   settings.RewardTrueNegative * traces.Count(t => !t.IsAccepting);
+        double fitnessLowerBound = 0.95 * fitnessUpperBound;
```
Comment (Sarmisuper) [inline GeneticDFA/Setup.cs:24]: this should be a parameter. Just the percentage

File: GeneticDFA/Utility/Settings.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+﻿namespace GeneticDFA.Utility;
+
+public class Settings
+{
+    public int MinPopulation { get; set; }
+    public int MaxPopulation { get; set; }
+    public int ConvergenceGenerationNumber { get; set; }
+    public int MaximumGenerationNumber { get; set; }
+    public int EliteSelectionScalingFactor { get; set; }
+    public int NumberOfFittestIndividualsAcrossAllGenerations => Convert.ToInt32(0.05 * MinPopulation);
```
Comment (Sarmisuper) [inline GeneticDFA/Utility/Settings.cs:10]: Percentage should be a parameter

File: GeneticDFA/Utility/Settings.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        ConvergenceGenerationNumber = 100;
+        MaximumGenerationNumber = 400;
+        EliteSelectionScalingFactor = 2;
+        RewardTruePositive = 10;
+        RewardTrueNegative = 10;
+        PenaltyFalsePositive = 10;
+        PenaltyFalseNegative = 10;
+        WeightNonDeterministicEdges = 2;
+        WeightUnreachableStates = 5;
+        WeightSize = 5;
```
Comment (Sarmisuper) [inline GeneticDFA/Utility/Settings.cs:45]: Sæt disse til 1

File: GeneticDFA/Utility/Settings.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        MutationProbability = 0.5;
+        NonDeterministicBehaviorProbability = 0.5;
```
Comment (Sarmisuper) [inline GeneticDFA/Utility/Settings.cs:47]: Sæt til 0,65

File: GeneticDFAUI/ViewModels/SettingsViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+    private int _eliteSelectionScalingFactor = 2;
+    private int _rewardTruePositive = 10;
+    private int _rewardTrueNegative = 10;
+    private double _penaltyFalsePositive = 10;
+    private double _penaltyFalseNegative = 10;
+    private double _weightNonDeterministicEdges = 2;
+    private double _weightUnreachableStates = 5;
+    private double _weightSize = 5;
+    private double _mutationProbability = 0.5;
+    private double _nonDeterministicBehaviorProbability = 0.5;
```
Comment (Sarmisuper) [inline GeneticDFAUI/ViewModels/SettingsViewModel.cs:28]: change these to the same as the other place

File: GeneticDFAUI/ViewModels/SettingsViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+        EliteSelectionScalingFactor = 2;
+        RewardTruePositive = 10;
+        RewardTrueNegative = 10;
+        PenaltyFalsePositive = 10;
+        PenaltyFalseNegative = 10;
+        WeightNonDeterministicEdges = 2;
+        WeightUnreachableStates = 5;
+        WeightSize = 5;
+        MutationProbability = 0.5;
+        NonDeterministicBehaviorProbability = 0.5;
```
Comment (Sarmisuper) [inline GeneticDFAUI/ViewModels/SettingsViewModel.cs:227]: samesies

**Review by Sarmisuper — CHANGES_REQUESTED** [review]

File: GeneticDFAUI/App.axaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:local="using:GeneticDFAUI"
+             x:Class="GeneticDFAUI.App">
+
+    <Application.DataTemplates>
+        <local:ViewLocator />
+    </Application.DataTemplates>
+
+    <Application.Styles>
+        <FluentTheme Mode="Light" />
```
[the remark this answers is not in the document]
Comment (KarmaKamikaze) [reply GeneticDFAUI/App.axaml:11]: Haha, jeg prøvede dark mode, men det skulle dælme tweakes for at blive pænt, så det blev ikke lige denne gang!

**Review by Sarmisuper — APPROVED** [review]

**Review by selectionator — APPROVED** [review]
LGTM, ISÆR Peekaboo 👍

**Pull request merged**